lambda/printlambda.ml

```diff
@@ -603,11 +603,9 @@ let rec lam ppf = function
   | Lfunction{kind; params; return; body; attr; mode; region} ->
       let pr_params ppf params =
         match kind with
-        | Curried {nlocal} ->
-            let first_local = List.length params - nlocal in
-            List.iteri (fun i (param, k) ->
-                fprintf ppf "@ %a%a%s" Ident.print param value_kind k
-                  (if first_local <= i then "[->L]" else "")) params
+        | Curried _ ->
+            List.iter (fun (param, k) ->
+                fprintf ppf "@ %a%a" Ident.print param value_kind k) params
```

Comment:
Any particular reason for removing this?

Reply:
It shows up in loads of tests, which would all need to have different versions of their output for the no stack allocation configuration. The information doesn't seem that useful so it seems easier to stop generating it. It also seems slightly better not to have this output in e.g. ocaml-ir where I suspect it will confuse anyone who isn't us.